### PR TITLE
Add voice transport and TwiML application.

### PR DESCRIPTION
**Pull request by hodgestar**

### CODE REVIEW

File: docker-vms/vxfreeswitch/vxfreeswitch.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+tts_fs_engine: "flite"
+tts_fs_voice: "kal"
+
+# middleware configuration
+
+middleware:
+  - logging_mw: vumi.middleware.logging.LoggingMiddleware
+
+logging_mw:
+  log_level: debug
```
Comment (smn) [inline docker-vms/vxfreeswitch/vxfreeswitch.yaml:34]: We probably want to `COPY` this file from the repo to the Docker image as part of the `Dockerfile` build recipe?

Comment (smn) [inline docker-vms/vxfreeswitch/vxfreeswitch.yaml:34]: Or is this where the environment variables come in?

Comment (hodgestar) [inline docker-vms/vxfreeswitch/vxfreeswitch.yaml:34]: We do want to copy this file (it's meant to contain all the settings that can be made public). The environment variables are only used in the Dockerfile CMD for the moment (they're not available at build time anyway I don't think).

Comment (hodgestar) [inline docker-vms/vxfreeswitch/vxfreeswitch.yaml:34]: Won't `COPY . /app` copy this file?

File: docker-vms/vxfreeswitch/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+FROM ubuntu
+MAINTAINER Praekelt Foundation <dev@praekeltfoundation.org>
+
+EXPOSE 8000
```
Comment (rudigiesler) [inline docker-vms/vxfreeswitch/Dockerfile:4]: Do we not also need to expose 8021 for freeswitch to connect to?

File: docker-vms/vxtwinio/vxtwinio.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# A vxtwinio application.
+
+# application configuration
+
+worker_name: vxtwinio_application
+transport_name: vxfreeswitch_transport
+
+web_path: "/vxtwinio"
+web_port: 8000
```
Comment (rudigiesler) [inline docker-vms/vxtwinio/vxtwinio.yaml:9]: Won't this clash with the transport on port 8000? Or are these on different machines?

**Comment by rudigiesler** [comment]
Just some comments about ports that are confusing me. Everything else looks good.

File: docker-vms/vxtwinio/vxtwinio.yaml (hunk 3 above)
Comment (hodgestar) [inline docker-vms/vxtwinio/vxtwinio.yaml:9]: They're different containers with different local network spaces, so they
can each have their own port 8000 (go containers!).

File: docker-vms/vxfreeswitch/Dockerfile (hunk 2 above)
Comment (hodgestar) [inline docker-vms/vxfreeswitch/Dockerfile:4]: The 8000 is mean to be for FreeSWITCH to connect to. Maybe I should just
change it to 8021 so it matches the default.

**Comment by rudigiesler** [comment]
:+1:

File: docker-vms/vxfreeswitch/vxfreeswitch.yaml (hunk 1 above)
Comment (smn) [inline docker-vms/vxfreeswitch/vxfreeswitch.yaml:34]: Doh. You're right. Sorry I missed that.

File: docker-vms/vxfreeswitch/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+COPY . /app
+WORKDIR /app
+
+RUN pip install vxfreeswitch
+
+CMD [ \
+    "twistd", "-n", "--pidfile=vxfreeswitch.pid", \
+    "vumi_worker", "--worker-class=vxfreeswitch.VoiceServerTransport", \
+    "--config=vxfreeswitch.yaml", \
+    "--set-option=freeswitch_auth:$FREESWITCH_AUTH" \
```
Comment (smn) [inline docker-vms/vxfreeswitch/Dockerfile:15]: :heart: the `$FREESWITCH_AUTH` approach here.

File: docker-vms/vxfreeswitch/Dockerfile (hunk 2 above)
Comment (smn) [inline docker-vms/vxfreeswitch/Dockerfile:4]: :+1: on making it the default, these ports are always re-mapped anyway with `docker run -p` for access outside the container.

**Pull request merged**